### DISCUSSION TITLE
Exit handler for derived app class?

### DISCUSSION THREAD

**Discussion by evanpolster** [Textualize/textual#3838] [Q&A]
(Apologies if this is a general Python question versus a specific Textual one.)
 
I have a Textual program with an my own derived App class.

I would like to wrap some cleanup code in a function inside my derived App class.

I notice that app.py has an _on_exit_app but it is asynchronous and I am not certain how or if to overload this.

I might attempt this:

```
class App:
    async def _on_exit_app(self) -> None:
        # Original implementation in the parent class
        self._begin_batch()
        await self._message_queue.put(None)

class SubApp(App):
    async def _on_exit_app(self) -> None:
        # Additional cleanup in the subclass
        # ... my cleanup logic ...

        # Then call the parent class's _on_exit_app
        await super()._on_exit_app()

```
What does not make sense in this crude example is that I am calling await twice. So it is "blocking" twice?

Is this the correct way to write exit handling in an App or is there a better way?

--Evan

**Comment by willmcgugan** [comment]
You could handle the unmount message. I.e. add `def on_unmount(self):`

**Reply by evanpolster** [reply]
Let me try that. Thanks!

**Reply by evanpolster** [reply]
That is the better solution. Thanks!
